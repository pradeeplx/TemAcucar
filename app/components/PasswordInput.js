import React, { Component, PropTypes } from 'react'
import FormTextInput from "./FormTextInput"

export default class PasswordInput extends Component {
  static propTypes = {
    value: PropTypes.string,
    valid: PropTypes.bool,
  }

  focus() {
    this.refs.input.focus()
  }

  blur() {
    this.refs.input.blur()
  }

  render() {
    return (
      <FormTextInput
        ref='input'
        name='password'
        title='Senha'
        placeholder='Digite sua senha'
        icon='ios-key-outline'
        autoCorrect={false}
        autoCapitalize='none'
        secureTextEntry={true}
        { ...this.props }
      />
    )
  }
}
