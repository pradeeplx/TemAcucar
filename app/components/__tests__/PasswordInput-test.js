import { View } from 'react-native'
import React from 'react'
import PasswordInput from '../PasswordInput'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <PasswordInput value='foo' valid={true} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});
