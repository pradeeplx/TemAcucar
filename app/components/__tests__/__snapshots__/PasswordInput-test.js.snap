exports[`test renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderBottomWidth": 0.5,
        "borderColor": "lightgray",
        "borderTopWidth": 0.5,
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#EFEDED",
        },
        undefined,
      ],
    ]
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingBottom": 6,
        "paddingTop": 6,
      }
    }>
    <View
      style={
        Array [
          Object {
            "marginLeft": 6,
          },
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#2CB36E",
              "fontFamily": "Ionicons",
              "fontSize": 27,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Array [
              Object {},
              undefined,
            ],
          ]
        }>
        ïŠ¼
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 0.5,
            "fontSize": 14,
            "fontWeight": "bold",
            "marginLeft": 6,
          },
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            undefined,
          ],
        ]
      }>
      Senha
    </Text>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      clearButtonMode="while-editing"
      icon={
        <Icon
          color="#654C50"
          name="ios-key-outline" />
      }
      iconStyle={Object {}}
      inputStyle={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#FF7F85",
            "fontSize": 18,
            "height": 54,
          },
          undefined,
        ]
      }
      invalidIcon={
        <Icon
          color="#E6D3B7"
          name="ios-more" />
      }
      message={null}
      messageStyle={
        Array [
          Object {
            "color": "#E6D3B7",
            "fontSize": 15,
          },
          undefined,
        ]
      }
      name="password"
      onBlur={undefined}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Digite sua senha"
      returnKeyType="done"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "fontSize": 14,
            "height": 36,
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "color": "#FF7F85",
              "fontSize": 18,
              "height": 54,
            },
            undefined,
          ],
        ]
      }
      title="Senha"
      titleStyle={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          undefined,
        ]
      }
      valid={true}
      validIcon={
        <Icon
          color="#2CB36E"
          name="md-checkmark" />
      }
      value="foo" />
  </View>
</View>
`;
